opm/parser/eclipse/EclipseState/Schedule/Completion.cpp

```diff
@@ -136,12 +138,22 @@ namespace Opm {
        disentangled, and each completion is returned separately.
     */
 
-    std::pair<std::string , std::vector<CompletionPtr> > Completion::completionsFromCOMPDATRecord( const DeckRecord& compdatRecord ) {
+    inline std::vector< CompletionPtr >
+    fromCOMPDAT( const DeckRecord& compdatRecord, const Well& well ) {
+
         std::vector<CompletionPtr> completions;
-        std::string well = compdatRecord.getItem("WELL").getTrimmedString(0);
+
         // We change from eclipse's 1 - n, to a 0 - n-1 solution
-        int I = compdatRecord.getItem("I").get< int >(0) - 1;
-        int J = compdatRecord.getItem("J").get< int >(0) - 1;
+        // I and J can be defaulted with 0 or *, in which case they are fetched
+        // from the well head
+        const auto& itemI = compdatRecord.getItem( "I" );
+        const auto defaulted_I = itemI.defaultApplied( 0 ) || itemI.get< int >( 0 ) == 0;
```

Comment:
Weak phrasing on my part. Here's a slightly more elaborate justification:

* `defaulted_I` should make it obvious that the boolean expression on the right hand side are the two conditions that would lead to this field being considered defaulted.
* The ternary on the `defaulted` should make it clear that the one case is behaviour in the case of defaulted, the other is when you simply read the value.
* Most important, it doesn't leave the question "why would you conditionally re-assign a value you've already read" open (the "why" I was referring to).
* And my not-so-humble perspective: I definitely don't have to scan the rest of the function for *more* possible reassignments, I know that we're dealing with a value.